src/vs/base/browser/ui/grid/grid.ts

```diff
@@ -577,21 +626,74 @@ export class Grid<T extends IView = IView> extends Disposable {
 	}
 
 	/**
-	 * Maximize the size of a {@link IView view} by collapsing all other views
+	 * Maximizes the specified view and hides all other views.
+	 * @param view The view to maximize.
+	 */
+	maximizeView(view: T): void {
+		if (this.views.size < 2) {
+			throw new Error('At least two views are required to maximize a view');
+		}
+		if (this.maximizedView === view) {
+			return;
+		}
+		if (this.maximizedView) {
+			this.unmaximizeView();
+		}
+
+		for (const otherView of this.views.keys()) {
+			if (view !== otherView) {
+				this.setViewVisible(otherView, false);
+			}
+		}
+
+		this.maximizedView = view;
+		this._onDidMaximizeGroup.fire({ view, maximized: true });
+	}
+
+	/**
+	 * Restores all views to their previous size after a view has been maximized.
+	 */
+	unmaximizeView(): void {
```

Comment:
Since we're not providing any view to `unmaximizeView`, it feels like a strange API method. It's implicit that it's only useful to unmaximize the currently maximized view. Thus a better name would potentially be `unmaximizeMaximizedView` which is ugly. Also `unmaximize` isn't a real word...

So why not just make `maximizeView` return an `IDisposable`?

Reply:
I gave this a try, but was not happy with it and so reverted it again. 
First of all, if we return a disposable, a reference to it has to be stored on both the `gridView` level and the `EditorPart` level as both levels can influence the maximized state. Secondly, when restoring the the grid after reloading the window, the disposable has to be reset somehow. This means we need to check if the active view is maximized, and then call the `maximizeView` method only to get the disposable.
So what I ended up doing is changing the name of the API method from `unmaximizeView()` to `exitMaximizedView()` which I find better.